stdlib/src/collections/dict.mojo

```diff
@@ -216,6 +216,15 @@ struct DictEntry[K: KeyElement, V: CollectionElement](
         self.key = other.key
         self.value = other.value
 
+    fn reap_value(owned self) -> V:
```

Comment:
Should we make this private? I don't expect users to need to interact with this API directly, right?

Reply:
hmm I'm not sure actually.
```mojo
def main():
    print(str(take()))

def take() -> String:
    var entry = DictEntry[String, String]("a", "b")
    return entry.value^
```
Doing this still causes a segfault and the language server to crash.. do you know if this is a bug, or maybe it should give me an error in the editor?

Reply:
As things are right now, it seems useful to have the function available and documented.

Reply:
Adding ```__mlir_op.`lit.ownership.mark_destroyed`(__get_mvalue_as_litref(entry))``` after `entry` is the solution to it, but that's definitely a private function.

Reply:
No, I don't think this should be private right now.